File: backend/util/size.go
package util

import "fmt"

// FormatBytes formats a size for human-readable logs (binary SI).
func FormatBytes(n int64) string {
	if n < 0 {
		n = 0
	}
	const unit = 1024
	if n < unit {
		return fmt.Sprintf("%d B", n)
	}
	div := int64(unit)
	exp := 0
	for n/div >= unit && exp < 4 {
		div *= unit
		exp++
	}
	val := float64(n) / float64(div)
	suffix := []string{"KiB", "MiB", "GiB", "TiB"}[exp]
	return fmt.Sprintf("%.2f %s", val, suffix)
}

File: frontend/src/utils/size.ts
/** Human-readable file size (binary SI), aligned with backend util.FormatBytes. */
export function formatBytes(n: number): string {
  if (!Number.isFinite(n) || n < 0) n = 0;
  const unit = 1024;
  if (n < unit) return `${n} B`;
  let div = unit;
  let exp = 0;
  while (n / div >= unit && exp < 4) {
    div *= unit;
    exp++;
  }
  const val = n / div;
  const suffix = ["KiB", "MiB", "GiB", "TiB"][exp];
  return `${val.toFixed(2)} ${suffix}`;
}
